### ISSUE TITLE
[TRIGGER] Hookdeck - New Event Received - Use async options for pre-existing sources

### ISSUE THREAD

**Issue by ctrlaltdylan** [PipedreamHQ/pipedream#10118]
**Describe the event source. What app is this for, and what event does the trigger correspond to?**

App: Hookdeck
Source: `new-event-received`

The issue is that when setting up the source, the `source_id` prop is an integer without a easy dropdown async option to select from.

This is inconvienent for a Hookdeck user setting up a new Hookdeck connection on Pipedream.

~Please enhance this prop so that an async option is used to create a dropdown of Hookdeck sources to select from the currently logged in account.~ Async options already exist. Thanks @michelle0927 !

Please also consider changing the prop label from `Source ID` to `existing Source`, and changing the label of `Source` to `New Source`, so it's more clear what the props actually do.

**Please provide a link to the relevant API docs for the specific service / operation this trigger is tied to.**

https://hookdeck.com/docs/api#retrieve-all-sources

**Comment by corners2wall** [comment]
@ctrlaltdylan  Can I take this?
By the way, I am newcomer in this repo. It's sound like not hard

**Comment by ctrlaltdylan** [comment]
Hi @corners2wall , feel free to make a PR! We're always happy to accept PRs on these issues.

**Comment by michelle0927** [comment]
@ctrlaltdylan It looks like the async options are already there?

<img width="564" alt="Screenshot 2024-01-30 at 12 14 39 PM" src="https://github.com/PipedreamHQ/pipedream/assets/10889229/5b15e2ee-4149-4524-8aab-1675c45c48f5">

**Comment by ctrlaltdylan** [comment]
@michelle0927 sorry, you're totally right.

Let me rephrase the asks:

1. Relabel `Source ID` to `Pre-exisiting Source` so it's more clear of what that prop does
2. Relabel `Source` to `New Source` so it's more clear that prop creates a new source
3. If possible, pass a `default` object (`default: { name: "My New Source" }`) to the `New Source` prop so that way the `name` field is already populated since that's the only required field (saves the user a click by looknig into the linked hookdeck docs)
